source/core/slang-io.h

```diff
@@ -25,8 +25,39 @@ namespace Slang
 	class Path
 	{
 	public:
+
+        enum class Type
+        {
+            Unknown,
+            File,
+            Directory,
+        };
+
+        typedef uint32_t TypeFlags;
+        struct TypeFlag
+        {
+            enum Enum : TypeFlags
+            {
+                Unknown = TypeFlags(1) << int(Type::Unknown),
+                File = TypeFlags(1) << int(Type::File),
+                Directory = TypeFlags(1) << int(Type::Directory),
+            };
+        };
+
+        class Visitor
+        {
+        public:
+            virtual void accept(Type type, const UnownedStringSlice& filename) = 0;
+        };
```

Comment:
A simple callback function that takes a "user data" pointer would seem to be simpler here, and can easily be wrapped with a template variant that adapts a lambda, allowing the use site to be very similar to a loop.